includes/functions/shared.php

```diff
@@ -112,6 +112,62 @@ function simcal_get_calendar( $object ) {
 	$objects = \SimpleCalendar\plugin()->objects;
 	return $objects instanceof \SimpleCalendar\Objects ? $objects->get_calendar( $object ) : null;
 }
+/**
+ * Formats strings to ical standard
+ *
+ * @param string $s
+ *
+ * @return string
+ */
+function format_ical_string( $s ) {
+    $r = wordwrap(
+        preg_replace(
+            array( '/,/', '/;/', '/[\r\n]/' ),
+            array( '\,', '\;', '\n' ),
+            $s
+        ), 73, "\n", TRUE
+    );
+
+    // Indent all lines but first:
+    $r = preg_replace( '/\n/', "\n  ", $r );
+
+    return $r;
+}
+
+
+/**
+ * Compose and return a ics feed from the calendar
+ * @param  string|int|object|WP_Post $object
+ * @return void
+ *
+ */
+function simcal_draw_calendar_ics($post) {
+	$calendar = simcal_get_calendar($post);
+	$events = $calendar->events;
+	header('Content-Type:text/calendar');
+	$ics = "BEGIN:VCALENDAR\r\nVERSION:2.0\r\nPRODID:-//Wordpress/Simple-Calendar\r\n";
+
+	foreach($events as $eventarr) {
+		$event = $eventarr[0];
+		if(strpos($ics, $event->uid) == false) {
+			$ics .= "BEGIN:VEVENT\r\n";
+			$ics .="UID:" . $event->uid . "\r\n";
+			$ics .="DTSTAMP:" . $event->start_dt->format('Ymd\THis') . "\r\n";
```

Comment:
I noticed that the timings are not matching, maybe because of timezone

<img width="249" alt="Screenshot 2023-03-20 at 8 37 03 AM" src="https://user-images.githubusercontent.com/3481438/226237373-79f6c4df-eed9-454c-8b0e-fc16995a9fd7.png">

(purple is ICS export and yellow is Google calendar)